src/lib/io/ABC.cpp

```diff
@@ -0,0 +1,429 @@
+                                interp == "point")
+                            {
+                                ptype = VECTOR;
+                            }
+                            oAttrMap[keyName] = ioData->addAttribute(
+                                propName.c_str(), ptype, extent);
+                        }
+                    }
+
+                    case Alembic::Util::kStringPOD:
+                    {
+                        if (oAttrMap.find(keyName) == oAttrMap.end())
+                        {
+                            oAttrMap[keyName] = ioData->addAttribute(
+                                propName.c_str(), INDEXEDSTR, extent);
+                        }
+                    }
+
+                    default:
+                    {
+                        continue;
+                    }
+
+                }
+            }
+        }
+    }
+
+    for (std::size_t i = 0; i < iObj.getNumChildren(); ++i)
+    {
+        Abc::IObject child = iObj.getChild(i);
+        findPoints(child, iTime, oNumPts, oAttrMap, ioData);
+    }
+}
+
+void writePoints(Abc::IObject & iObj, double iTime,
+                const std::map<std::string, ParticleAttribute> & iAttrMap,
+                ParticlesDataMutable * iData,
+                uint64_t & oIndex,
+                std::vector<Imath::M44d> & oMatStack)
+{
+
+    bool shouldPop = false;
+
+    if (AbcGeom::IXform::matches(iObj.getMetaData()))
+    {
+        AbcGeom::IXform xformObj(iObj.getParent(), iObj.getName());
+        AbcGeom::IXformSchema schema = xformObj.getSchema();
+        AbcGeom::XformSample samp;
+
+        // should we need to linearly interpolate?
+        schema.get(samp, Alembic::Abc::ISampleSelector(iTime));
+        Abc::M44d mat = samp.getMatrix();
+
+        // doesnt inherit? just push this matrix on the stack
+        if (!samp.getInheritsXforms())
+        {
+            shouldPop = true;
+            oMatStack.push_back(mat);
+        }
+        else if (mat != Abc::M44d())
+        {
+            shouldPop = true;
+            oMatStack.push_back(mat * oMatStack[oMatStack.size() - 1]);
+        }
+    }
+    else if (AbcGeom::IPoints::matches(iObj.getMetaData()))
+    {
+        Abc::ISampleSelector sel(iTime);
+        Abc::ICompoundProperty geomProp(iObj.getProperties(), ".geom");
+        Abc::ICompoundProperty arbGeomProp;
+        if (geomProp.getPropertyHeader(".arbGeomParams") != NULL)
+        {
+            arbGeomProp = Abc::ICompoundProperty(geomProp, ".arbGeomParams");
+        }
+
+        std::size_t numPts = 0;
+
+        Abc::IArrayProperty pProp(geomProp, "P");
+        if (pProp.getNumSamples() > 0)
+        {
+            Alembic::Util::Dimensions dims;
+            pProp.getDimensions(dims, sel);
+            numPts = dims.numPoints();
+        }
+
+        std::map<std::string, ParticleAttribute>::const_iterator it;
+        for (it = iAttrMap.begin(); it != iAttrMap.end(); ++it)
+        {
```

Comment:
Tiny style sug ~ map iterators are one of those areas where `auto` is kind of nice. Could we even go one step further and use a new-style loop here?

```
for (const auto& entry : iAttrMap) {
   // entry.first, entry.second
}
```